### PR TITLE
Incorrect metadata for sql_variant datatype on a restored server

### PR DESCRIPTION
**Pull request by aadityavermaa**
### Description

When pg_dump/restore is performed on database with table containing sql_variant data type columns, it was observed that the metadata seemed to get lost during the dump process. Therefore, the values were not correctly restored, which reflected in the failure of some test cases in JDBC framework. This change aims to separately handle table containing sql_variant data type columns so that the values can be correctly dumped and restored. 

Testing is covered by existing tests in `babel_datatype_sqlvariant` and other tests for sql_variant datatype (e.g. `TestSqlVariant`, `TestNotNull` and other tests in `/input/sql_variant` folder ). More test scenarios are included in this change such as generated columns and basetypes such as nchar, char, varchar, numeric, decimal, etc. 

Engine changes: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/131
Signed-off-by: Aditya Verma <adiityav@amazon.com>
### Issues Resolved

BABEL-4036

### Test Scenarios Covered ###

* **Use case based -** Yes

* **Boundary conditions -** Yes

* **Arbitrary inputs -** Yes

* **Negative test cases -** Yes

* **Minor version upgrade tests -** Yes

* **Major version upgrade tests -** Yes

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rishabhtanwar29** [comment]
Let's add tests for tables with generated columns before/after sql_variant column, generated columns dependent upon sqlvariant column etc.

File: test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out [hunk 1]
```diff
@@ -753,3 +753,86 @@ insert into babel_datatype_sqlvariant_vu_prepare_t9 values (cast('$200' as money
...
+go
+~~ROW COUNT: 1~~
+
+
+insert into babel_4036_t1 values (6, cast('2016-10-23 12:45:37.1234567+10:0' as datetime2), 'String 6', cast('2016-10-23 12:45:37.1234567 +10:0' as datetime2(5)), cast('2016-10-23 12:45:37.1234567 +10:0' as datetime2(7)));
+go
+~~ROW COUNT: 1~~
+
+
+insert into babel_4036_t1 values (8, cast(1234.56789 as numeric(7,2)), 'String 8', cast(1234.56789 as numeric(9,3)), cast(cast(1234.56789 as numeric(9,5))));
```
Comment (surendravishnoi) [inline test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out:790]: for numeric and decimal, as we are inferring scale and precision, let us add detailed testing with cases like max precision, 0 scale and negative/positive and so on

File: test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out [hunk 2]
```diff
@@ -753,3 +753,86 @@ insert into babel_datatype_sqlvariant_vu_prepare_t9 values (cast('$200' as money
...
+
+create table babel_4036_t1( a int, b sql_variant, c varchar(50), d sql_variant, e sql_variant);
+go
+
+insert into babel_4036_t1(a,c,e) values (1, 'String 1', cast('Varchar String' as varchar(50)));
+go
+~~ROW COUNT: 1~~
+
+
+insert into babel_4036_t1 values (2, cast('😊😋😎😍😅😆' as nchar(24)), 'String 2', cast('😊😋😎😍😅😆' as nchar(26)),cast('😊  😋 😎dsfsd😍 😅ds😆' as nchar(24)));
```
Comment (surendravishnoi) [inline test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out:765]: we should have test case for each type we are handling, char, nvarchar, binary, varbinary, money bit and so on

File: test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out (hunk 1 above)
Comment (aadityavermaa) [reply test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out:790]: Added some tests.

File: test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out (hunk 2 above)
Comment (aadityavermaa) [reply test/JDBC/expected/babel_datatype_sqlvariant-vu-prepare.out:765]: There are other tests files for sql_variant datatype that already contain tests for different datatypes, including this file and others ( e.g. - TestSqlVariant, TestNotNull, other tests in input/sql_variant folder )

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by surendravishnoi — APPROVED** [review]

**Pull request merged**